src/pykeen/training/training_loop.py

```diff
@@ -524,7 +525,18 @@ def _train(  # noqa: C901
         if sampler == "schlichtkrull":
             if triples_factory is None:
                 raise ValueError("need to pass triples_factory when using graph sampling")
-            sampler = GraphSampler(triples_factory, num_samples=sub_batch_size)
+            if not isinstance(training_instances, SLCWAInstances):
+                raise NotImplementedError("Subgraph sampling is currently only supported for SLCWA training.")
+            # wrap training instances
+            training_instances = SLCWASubGraphInstances(
+                mapped_triples=triples_factory.mapped_triples,
+                sub_graph_size=sub_batch_size,
+            )
+            # disable automatic batching
+            batch_size = None
+            # no support for sub-batching
+            sub_batch_size = None
+            sampler = None
```

Comment:
you might need to explicitly set drop_last to false if using schlichtkrull sampling

Reply:
f5f6b4f